src/GovUk.Education.ExploreEducationStatistics.Common/Cache/CacheAttribute.cs

```diff
@@ -74,8 +77,348 @@ private void ValidateOptions()
             }
         }
 
-        public abstract Task<object?> Get(ICacheKey cacheKey, Type returnType);
+        protected override T WrapSync<T>(Func<object[], T> target, object[] args, AspectEventArgs eventArgs)
+        {
+            var unboxedResultType = typeof(T).GetUnboxedResultTypePath().Last();
+
+            var cacheKey = GetCacheKey(Key, args, eventArgs.Method);
+
+            object? cachedResult = null;
+            if (!ForceUpdate)
+            {
+                cachedResult = Get(cacheKey, unboxedResultType);
+            }
+
+            if (cachedResult?.TryBoxToResult(typeof(T), out cachedResult) == true)
+            {
+                return (T)cachedResult;
```

Comment:
This cast previously wasn't necessary because we were returning inside the handle method, which has the return type `object`. Now that this code is inside a WrapAsync/WrapSync methods, which return `T`, we now need the cast.

I think to fix this we need the out param type of `TryBoxToResult` to be dynamic, depending on what type is passed as an argument to it, and I don't believe we can do that.

If there is a way to avoid this cast, I'm not sure what it is. `TryBoxToResult` checks that `cachedResult` is an instance of type `T`, but I guess we could also do this check in WrapSync/WrapAsync to protect against future refactors of `TryBoxToResult`, but I'm inclined to think it's `TryBoxToResult`'s unit tests' responsibility to protect against this. So I'm personally inclined to leave this.